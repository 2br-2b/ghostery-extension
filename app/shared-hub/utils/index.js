/**
 * React Utility Imports
 * Import files from app to prevent duplicate code
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Imports utilities from elsewhere in the codebase to reduce duplicate code
import {
	applyMiddleware,
	compose,
	combineReducers,
	createStore
} from 'redux';
import thunk from 'redux-thunk';

import { log } from '../../../src/utils/common';
import { sendMessage as importedSM, sendMessageInPromise as importedSMIP } from '../../panel/utils/msg';

const sendMessageInPromise = function(name, message) {
	return importedSMIP(name, message, 'ghostery-hub');
};

const sendMessage = function(name, message) {
	return importedSM(name, message, 'ghostery-hub');
};

/**
 * Build store using combined reducers and middleware
 * @return {Object}
 * @memberof HubReactStore
 */
const createStoreFactory = function(reducers) {
	const reducer = combineReducers(reducers);

	return createStore(
		reducer,
		compose(
			applyMiddleware(thunk),
			window.devToolsExtension ? window.devToolsExtension() : f => f
		),
	);
};

export { createStoreFactory, log, sendMessage, sendMessageInPromise };
