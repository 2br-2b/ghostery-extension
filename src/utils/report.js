/**
 * Comparing performance of webRequest handlers
 * of Ghostery, adblocker and antitracking
*/
console.log("AVAILABLE?", typeof self.localStorage, typeof self.indexedDB, typeof self.chrome, typeof self.storage);
let TOTAL_GH = 0;
let TOTAL_AD = 0;
let TOTAL_AN = 0;

let PAGE_TOTAL_AD = 0;
let	PAGE_TOTAL_AN = 0;
let	PAGE_TOTAL_GH = 0;

let AD2G_MAX_RATIO = 0;
const AD2G_MAX_RATIO_TRESHOLD = 10;

let AN2G_MAX_RATIO = 0;
const AN2G_MAX_RATIO_TRESHOLD = 10;

let G2AD_MAX_RATIO = 0;
const G2AD_MAX_RATIO_TRESHOLD = 1;

let G2AN_MAX_RATIO = 0;
const G2AN_MAX_RATIO_TRESHOLD = 1;

let PAGE_URL = "";

let PAGE_LEVEL = true;
let REQUEST_LEVEL = false;

const AN_TOP = {duration: 0, url: ""};
const AD_TOP = {duration: 0, url: ""};
const GH_TOP = {duration: 0, url: ""};

//Ghostery handlers
const GH_HANDLERS = [
	'ghostery.onBeforeRequest',
	'ghostery.onHeadersReceived'
];
//Adblocker handlers
const AD_HANDLERS = [
	'adblocker'
];
//Antitracking handlers 
const AN_HANDLERS = [
'antitracking.onBeforeRequest',
'antitracking.onBeforeSendHeaders',
'antitracking.onHeadersReceived',
'redirectTagger.checkRedirect',
'pageLogger.logMainDocument',
'skipInvalidSource',
'skipInternalProtocols',
'checkSameGeneralDomain',
'cancelRecentlyModified',
'subdomainChecker.checkBadSubdomain',
'pageLogger.attachStatCounter',
'pageLogger.logRequestMetadata',
'checkExternalBlocking',
'tokenExaminer.examineTokens',
'tokenTelemetry.extractKeyTokens',
'domChecker.checkDomLinks',
'domChecker.parseCookies',
'tokenChecker.findBadTokens',
'checkSourceWhitelisted',
'checkShouldBlock',
'isQSEnabled',
'blockRules.applyBlockRules',
'applyBlock',

'cookieContext.assignCookieTrust',
'redirectTagger.confirmRedirect',
'checkIsMainDocument',
'skipInvalidSource',
'skipInternalProtocols',
'checkSameGeneralDomain',
'subdomainChecker.checkBadSubdomain',
'pageLogger.attachStatCounter',
'catchMissedOpenListener',
'overrideUserAgent',
'checkHasCookie',
'checkIsCookieWhitelisted',
'cookieContext.checkCookieTrust',
'cookieContext.checkVisitCache',
'cookieContext.checkContextFromEvent',
'shouldBlockCookie',
'blockCookie',

'antitracking.onHeadersReceived',
'checkMainDocumentRedirects',
'skipInvalidSource',
'skipInternalProtocols',
'skipBadSource',
'checkSameGeneralDomain',
'redirectTagger.checkRedirectStatus',
'pageLogger.attachStatCounter',
'logResponseStats',
'checkSetCookie',
'shouldBlockCookie',
'checkIsCookieWhitelisted',
'cookieContext.checkCookieTrust',
'cookieContext.checkVisitCache',
'cookieContext.checkContextFromEvent',
'blockSetCookie'
];
/**
 * Reset all flags and report results
 * once page url changes. If not - does not do anything
 * @param  {Object} data 
  */
function Reset(data) {
	let retVal;
	const {page_url} = data;
	if(page_url && (PAGE_URL !== page_url)) {
		console.log(`\r\nOLD PAGE: ${PAGE_URL} NEW PAGE: ${page_url}`);
		if(PAGE_LEVEL) {
			const INCREASE = (PAGE_TOTAL_GH + PAGE_TOTAL_AD + PAGE_TOTAL_AN)/PAGE_TOTAL_GH;
			retVal = { INCREASE, PAGE_TOTAL_GH, GH_TOP, PAGE_TOTAL_AD, AD_TOP, PAGE_TOTAL_AN, AN_TOP };
			console.log(`
GN_LATENCY: ${PAGE_TOTAL_GH}
GH_TOP: ${GH_TOP.duration}, ${GH_TOP.url}
AD_LATENCY: ${PAGE_TOTAL_AD}
AD_TOP: ${AD_TOP.duration}, ${AD_TOP.url}
AN_LATENCY: ${PAGE_TOTAL_AN}
AN_TOP: ${AN_TOP.duration}, ${AN_TOP.url}

Increase: ${INCREASE} times

`);
		}
		GH_TOP.duration = 0;
		GH_TOP.url = 0;

		AD_TOP.duration = 0;
		AD_TOP.url = 0;

		AN_TOP.duration = 0;
		AN_TOP.url = 0;

		PAGE_URL = page_url;
		TOTAL_AD = 0;
		TOTAL_AN = 0;
		TOTAL_GH = 0;
		PAGE_TOTAL_AD = 0;
		PAGE_TOTAL_AN = 0;
		PAGE_TOTAL_GH = 0;
		AD2G_MAX_RATIO = 0;
		AN2G_MAX_RATIO = 0;
		G2AD_MAX_RATIO = 0;
		G2AN_MAX_RATIO = 0;
	}

	return retVal;
}
/**
 * Calculates total time spent in a group of handlers 
 * belonging to different modules in tht time the page is loading 
 * @param  {string} name     handler name
 * @param  {string} url  	 url of the request
 * @param  {number} duration duration  in ms[description]
 */
function Compare(data) {
	const retVal = Reset(data);
	const {handler, url, duration} = data;

	if(GH_HANDLERS.includes(handler)) {
		PAGE_TOTAL_GH += duration; 
		if(GH_TOP.duration < duration) {
			GH_TOP.duration = duration;
			GH_TOP.url = url; 
		}
	} else if (AD_HANDLERS.includes(handler)) {
		PAGE_TOTAL_AD += duration; 
		TOTAL_AD += duration;
		if(AD_TOP.duration < duration) {
			AD_TOP.duration = duration;
			AD_TOP.url = url; 
		}

	} else if(AN_HANDLERS.includes(handler)) {
		PAGE_TOTAL_AN += duration; 
		TOTAL_AN += duration;
		if(AN_TOP.duration < duration) {
			AN_TOP.duration = duration;
			AN_TOP.url = url; 
		}
	} 

	return retVal;
}
const DB_NAME = 'perfDb';
const NAVIGATION_STORE = 'navigation';
//Delete database on startup
// const oldDb = self.indexedDB.deleteDatabase('perfDB');
// oldDb.onsuccess = function () {
//     console.log("Deleted database successfully");

	let db;
	const req = self.indexedDB.open(DB_NAME, 1);
	req.onsuccess = (e) => {
		console.log('successfully opened db', e);
		db = e.target.result;
	};
	req.onupgradeneeded = (e) => {
		console.log('successfully upgraded db', e);
		db = e.target.result;

		const objectStore = db.createObjectStore(NAVIGATION_STORE, { keyPath: 'page_url' });
	};
	req.onerror = (e) => {
		console.log('database error', e);
	};

	self.onmessage = (e) => {
		const {name} = e.data;
		if(name === 'perfData') {
			const retVal = Compare(e.data);
			if(retVal) {
				console.log("WRITING TO DATABASE");
				const {page_url} = e.data;
			 	const transaction = db.transaction(NAVIGATION_STORE, 'readwrite');
			 	const objectStore = transaction.objectStore(NAVIGATION_STORE);
			 	if(page_url) {
			 		const request = objectStore.get(page_url);
					request.onerror = function (event) {
					  console.log('GET ERROR!!!!!!!!!!');
					};
					request.onsuccess = function (event) {
					  // Get the old value that we want to update
						let dbData = event.target.result;
						if (!dbData || !dbData.dataArray) {
							dbData = { page_url: page_url, dataArray: [retVal] };
						} else {
							// update the value(s) in the object that you want to change
							dbData.dataArray.push(retVal);
						}

						// Put this updated object back into the database.
						const requestUpdate = objectStore.put(dbData);
						requestUpdate.onerror = function (event) {
						    console.log('PUT ERROR!!!!!!!!!!');
						};
						requestUpdate.onsuccess = function (event) {
							console.log("Successfully updated data with", page_url, retVal);
						};
					};
				} else {
					//console.log("NO PAGE URL", e);
				}
			};
		} else if(name === 'exportData') {
		    const trans = db.transaction(NAVIGATION_STORE, 'readonly');
		    const store = trans.objectStore(NAVIGATION_STORE);
		    const items = [];
		 
		    trans.oncomplete = function(evt) { 
		    	console.log("ITEMS", items);
		    	self.postMessage({origin: 'perf-worker', name:'exportPerfData', message: items});
		    };
		 
		    var cursorRequest = store.openCursor();
		 
		    cursorRequest.onerror = function(error) {
		        console.log('CURSOR ERROR', error);
		    };
		 
		    cursorRequest.onsuccess = function(evt) {                    
		        var cursor = evt.target.result;
		        if (cursor) {
		            items.push(cursor.value);
		            cursor.continue();
		        }
		    };
		}
	}
//};
// oldDb.onerror = function () {
//     console.log("Couldn't delete database");
// };
// oldDb.onblocked = function () {
//     console.log("Couldn't delete database due to the operation being blocked");
// };



