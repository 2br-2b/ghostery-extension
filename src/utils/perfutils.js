import * as utils from './utils';
import conf from '../classes/Conf';
import globals from '../classes/Globals';

// ID of Ghostery-perf extension
const RECEIVER_ID = 'pdlmemohjlhncchohlaeifdmbjbngcld';
let PORT;
chrome.runtime.onMessageExternal.addListener(
  function(request, sender, sendResponse) {
    if (sender.id === RECEIVER_ID) {
    	const {name} = request;
    	if(name === 'perfReady') {
    		sendResponse(_reportStaticInfo());
    		PORT = chrome.runtime.connect(RECEIVER_ID);
    		return true;
    	}
    }

    return false;
});

function _reportStaticInfo() {
	let gb;
	let hasOnes = false;
	let hasZeros = false;
	Object.keys(conf.selected_app_ids).forEach(key => {
		if(conf.selected_app_ids[key] === 1) {
			hasOnes = true;
		}
		if(conf.selected_app_ids[key] === 0) {
			hasZeros = true;
		}
	});
	gb = (hasOnes && hasZeros) ? 2 : hasOnes ? 1 : 0;
	return { name: 'perfInfo', 
		ghostery_version: chrome.runtime.getManifest().version,
		 
		at: conf.enable_anti_tracking ? "1" : "0",
		ab: conf.enable_ad_block ? "1" : "0",
		sb: conf.enable_smart_block ? "1" : "0",
		gb: gb,
		gp: globals.SESSION.paused_blocking 
	};
}

function _report(perfEntry, data) {
	if (PORT) {
		const report = {
			name: 'perfData', startTime: perfEntry.startTime, duration: perfEntry.duration, handler: data.handler, page_url: data.page_url, url: data.url
		};
		PORT.postMessage(report);
	}
}

/**
 * @namespace BackgroundUtils
 */
/**
 * Set the start mark
 * @param  {string} name name of the mark, should be unique
 */
export function perfBegin(name) {
	console.log('PERF BEGIN CALLED FOR', name);
	window.performance.mark(`BEGIN_${name}`);
}
/**
 * Set the end mark and measure the duration.
 * @param  {string} name name of the mark
 * @return {number}      duration
 */
export function perfEnd(name, data) {
	let entry = { name: `BEGIN_${name}`, startTime: 0, duration: 0 };
	const p = window.performance.getEntriesByName(`BEGIN_${name}`, 'mark') || [];
	if (p && p.length) {
		console.log('BEGIN MARK WAS FOUND FOR', name);
		window.performance.mark(`END_${name}`);
		window.performance.measure(name, `BEGIN_${name}`, `END_${name}`);

		const measures = window.performance.getEntriesByName(name, 'measure') || [];
		if (measures && measures.length) {
			entry = measures[0];
			if (entry && data) {
				data.handler = name;
				_report(entry, data);
			}
		} else {
			console.log('BAD MEASURE');
		}
	} else {
		console.log('BEGIN MARK WAS NOT FOUND FOR', name);
	}
	window.performance.clearMarks(`BEGIN_${name}`);
	window.performance.clearMarks(`END_${name}`);

	window.performance.clearMeasures(name);
	return entry;
}

