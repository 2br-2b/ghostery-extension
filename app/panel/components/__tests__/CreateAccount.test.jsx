import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { shallow } from 'enzyme';
import ClassNames from 'classnames';
import RSVP from 'rsvp';

import { validateEmail, validateConfirmEmail, validatePassword } from '../../utils/utils';
import CreateAccount from '../CreateAccount';

describe('app/panel/components/CreateAccount Component', () => {
	describe('Snapshot test with react-test-renderer', () => {
		test('Testing account creation page is rendering', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false,
			}

			const component = renderer.create(
				shallow(<MemoryRouter>
					<CreateAccount {...initialState} />
				</MemoryRouter>)	
			).toJSON(); 
			expect(component).toMatchSnapshot();
		});

		test('Testing the account creation page is rendering with the errors showing', () => {
			const initialState= {
				email: '',
				emailError: true,
				confirmEmail: '',
				confirmEmailError: true,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: false,
				loading: true,
				passwordInvalidError: true,
				passwordLengthError: true,
			}

			const component = renderer.create(
				shallow(<MemoryRouter>
					<CreateAccount {...initialState} />
				</MemoryRouter>)	
			).toJSON(); 
			expect(component).toMatchSnapshot();
		});
	});	

	describe('Testing the handling of data input and submitting', () => {
		test('testing the input of email and password without first/last name', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false,
			}	

			const wrapper = shallow(<CreateAccount {...initialState} />);

			const email = 'test@example.com';
			expect(wrapper.state('email')).toEqual('');
			wrapper.find('#create-input-email').simulate('change', {target: {name: 'email', value: email }});
			expect(wrapper.state('email')).toEqual(email);

			expect(wrapper.state('confirmEmail')).toEqual('');
			wrapper.find('#create-input-email-confirm').simulate('change', {target: {name: 'confirmEmail', value: email }});
			expect(wrapper.state('confirmEmail')).toEqual(email);

			expect(wrapper.state('promotionsChecked')).toBe(true);

			const password = 'admin';
			expect(wrapper.state('password')).toEqual('');
			wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: password }});
			expect(wrapper.state('password')).toEqual(password);
		});

		test('testing the input of email and password with first/last name', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false,
			}

			const wrapper = shallow(<CreateAccount {...initialState} />);

			const email = 'test@example.com';
			expect(wrapper.state('email')).toEqual('');
			wrapper.find('#create-input-email').simulate('change', {target: {name: 'email', value: email }});
			expect(wrapper.state('email')).toEqual(email);

			expect(wrapper.state('confirmEmail')).toEqual('');
			wrapper.find('#create-input-email-confirm').simulate('change', {target: {name: 'confirmEmail', value: email }});
			expect(wrapper.state('confirmEmail')).toEqual(email);

			const firstName = 'John';
			expect(wrapper.state('firstName')).toEqual('');
			wrapper.find('#create-input-first-name').simulate('change', {target: {name: 'firstName', value: firstName }});
			expect(wrapper.state('firstName')).toEqual(firstName);

			const lastName = 'Doe';
			expect(wrapper.state('lastName')).toEqual('');
			wrapper.find('#create-input-last-name').simulate('change', {target: {name: 'lastName', value: lastName }});
			expect(wrapper.state('lastName')).toEqual(lastName);

			expect(wrapper.state('promotionsChecked')).toBe(true);

			const password = 'admin';
			expect(wrapper.state('password')).toEqual('');
			wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: password }});
			expect(wrapper.state('password')).toEqual(password);
		});

		test('Creation of account with correct data', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false
			}				

			const wrapper = shallow(<CreateAccount {...initialState} />);
			const myMock = jest.fn();	
			wrapper.instance().handleSubmit = myMock;
			wrapper.instance().forceUpdate();

			wrapper.find('#create-input-email').simulate('change', {target: {name: 'email', value: 'test@example.com'}});
			wrapper.find('#create-input-email-confirm').simulate('change', {target: {name: 'confirmEmail', value: 'test@example.com'}});
			wrapper.find('#create-input-first-name').simulate('change', {target: {name: 'firstName', value: 'John'}});
			wrapper.find('#create-input-last-name').simulate('change', {target: {name: 'lastName', value: 'Doe'}});	
			wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'admin'}});			

			expect(wrapper.state('promotionsChecked')).toBe(true);
			expect(myMock.mock.calls.length).toBe(0);

			wrapper.find('form').simulate('submit');
			expect(myMock.mock.calls.length).toBe(1);

			expect(wrapper.state('emailError')).toBe(false);
			expect(wrapper.state('confirmEmailError')).toBe(false);
			expect(wrapper.state('passwordInvalidError')).toBe(false);
			expect(wrapper.state('passwordLengthError')).toBe(false);
		});

		test.only('testing the creation of account with checkbox unchecked', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false
			}				

			const wrapper = shallow(<CreateAccount {...initialState} />);
			const myMock = jest.fn();	
			wrapper.instance().handleSubmit = myMock;
			wrapper.instance().forceUpdate();

			wrapper.find('#create-input-email').simulate('change', {target: {name: 'email', value: 'test@example.com'}});
			wrapper.find('#create-input-email-confirm').simulate('change', {target: {name: 'confirmEmail', value: 'test@example.com'}});
			wrapper.find('#create-input-first-name').simulate('change', {target: {name: 'firstName', value: 'John'}});
			wrapper.find('#create-input-last-name').simulate('change', {target: {name: 'lastName', value: 'Doe'}});	
			wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'admin'}});			

			wrapper.find('#promotionsChecked').simulate('change', {target: {checked: false}});	
			expect(myMock.mock.calls.length).toBe(0);

			wrapper.find('form').simulate('submit');
			expect(myMock.mock.calls.length).toBe(1);
			expect(wrapper.state('promotionsChecked')).toBe(false);
		});

		test('testing the creation of account with invalid email', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false,
			}

			const wrapper = shallow( <CreateAccount {...initialState} />);
			wrapper.find('#create-input-email').simulate('change', {
					target: {
						name: 'email', 
						value: 'test1'
					}});
			wrapper.find('#create-input-email-confirm').simulate('change', {target: {name: 'confirmEmail', value: 'test1'}});
			wrapper.find('input[type="password"]').simulate('change', {
					target: {
						name: 'password', 
						value: 'hi'
					}});
			expect(wrapper.state('promotionsChecked')).toBe(true);

			const preventDefault = jest.fn();
			expect(preventDefault.mock.calls.length).toBe(0);


			wrapper.find('form').simulate('submit',  { preventDefault });
			expect(preventDefault.mock.calls.length).toBe(1);
		
			expect(wrapper.state('emailError')).toBe(true);
		});	


		test('testing the creation of account with invalid password', () => {
			const initialState= {
				email: '',
				emailError: false,
				confirmEmail: '',
				confirmEmailError: false,
				firstName: '',
				lastName: '',
				password: '',
				promotionsChecked: true,
				loading: false,
				passwordInvalidError: false,
				passwordLengthError: false,
			}

			const wrapper = shallow( <CreateAccount {...initialState} />);
			wrapper.find('#create-input-email').simulate('change', {
					target: {
						name: 'email', 
						value: 'test@example.com'
					}});
			wrapper.find('#create-input-email-confirm').simulate('change', {target: {name: 'confirmEmail', value: 'test@example.com'}});
			wrapper.find('input[type="password"]').simulate('change', {
					target: {
						name: 'password', 
						value: 'hi'
					}});
			expect(wrapper.state('promotionsChecked')).toBe(true);

			const preventDefault = jest.fn();
			expect(preventDefault.mock.calls.length).toBe(0);


			wrapper.find('form').simulate('submit',  { preventDefault });
			expect(preventDefault.mock.calls.length).toBe(1);
		
			expect(wrapper.state('passwordLengthError')).toBe(true);	
		});
	});	
});
