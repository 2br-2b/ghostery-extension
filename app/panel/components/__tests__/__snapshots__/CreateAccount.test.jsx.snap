// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel/components/CreateAccount Component Snapshot test with react-test-renderer Testing account creation page is rendering 1`] = `
<div
  id="create-account-panel"
>
  <div
    className="row align-center"
  >
    <div
      className="small-11 medium-8 columns"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="columns"
          >
            <div
              className=""
              id="create-account-email"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-email"
                id="create-email-label"
              >
                email_field_label
                <span
                  className="asterisk"
                >
                  *
                </span>
                <input
                  autoComplete="off"
                  className="create-account-input"
                  id="create-input-email"
                  name="email"
                  onChange={[Function]}
                  pattern=".{1,}"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
              <p
                className="warning"
                id="email-invalid"
              >
                invalid_email_create
              </p>
            </div>
            <div
              className=""
              id="create-account-email-confirm"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-email-confirm"
                id="create-email-confirm-label"
              >
                email_confirm_field_label
                <span
                  className="asterisk"
                >
                  *
                </span>
                <input
                  autoComplete="off"
                  className="create-account-input"
                  id="create-input-email-confirm"
                  name="confirmEmail"
                  onChange={[Function]}
                  pattern=".{1,}"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
              <p
                className="warning"
              >
                invalid_email_confirmation
              </p>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-6 columns"
          >
            <div
              id="create-account-first-name"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-first-name"
                id="create-first-name-label"
              >
                first_name_field_label
                <input
                  className="create-account-input"
                  id="create-input-first-name"
                  name="firstName"
                  onChange={[Function]}
                  pattern=".{1,}"
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="small-6 columns"
          >
            <div
              id="create-account-last-name"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-last-name"
                id="create-last-name-label"
              >
                last_name_field_label
                <input
                  className="create-account-input"
                  id="create-input-last-name"
                  name="lastName"
                  onChange={[Function]}
                  pattern=".{1,}"
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-12 columns"
          >
            <div
              className=""
              id="create-account-password"
            >
              <div
                className="row"
              >
                <div
                  className="columns"
                >
                  <label
                    className="create-account-label"
                    htmlFor="create-input-password"
                    id="create-password-label"
                  >
                    create_password_field_label
                    <span
                      className="asterisk"
                    >
                      *
                    </span>
                    <input
                      className="create-account-input"
                      id="create-input-password"
                      name="password"
                      onChange={[Function]}
                      pattern=".{1,}"
                      required={true}
                      type="password"
                      value=""
                    />
                  </label>
                </div>
              </div>
              <p
                className="warning"
              >
                <span
                  className=""
                  id="password-length-requirement"
                >
                  password_requirements
                </span>
                <span
                  className=""
                  id="password-characters-requirement"
                >
                  password_characters_requirements
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-12 columns"
          >
            <div
              id="create-account-promotions"
            >
              <input
                checked={true}
                id="promotionsChecked"
                name="promotionsChecked"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="promotionsChecked"
              >
                hub_create_account_checkbox_promotions
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-12 columns"
          >
            <div
              id="create-account-privacy-container"
            >
              <p
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "account_creation_privacy_statement",
                  }
                }
                id="accept-privacy-label"
              />
            </div>
            <div
              className="row align-center"
              id="account-creation-buttons"
            >
              <div
                className="small-6 columns text-center"
              >
                <a
                  className="cancel button hollow"
                  href="/"
                  id="create-account-cancel"
                  onClick={[Function]}
                >
                  button_cancel
                </a>
              </div>
              <div
                className="small-6 columns text-center"
              >
                <button
                  className="button ghostery-button"
                  id="create-account-button"
                  type="submit"
                >
                  <span
                    className="title"
                  >
                    panel_title_create_account
                  </span>
                  <span
                    className="loader"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`app/panel/components/CreateAccount Component Snapshot test with react-test-renderer Testing the account creation page is rendering with the errors showing 1`] = `
<div
  id="create-account-panel"
>
  <div
    className="row align-center"
  >
    <div
      className="small-11 medium-8 columns"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="columns"
          >
            <div
              className=""
              id="create-account-email"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-email"
                id="create-email-label"
              >
                email_field_label
                <span
                  className="asterisk"
                >
                  *
                </span>
                <input
                  autoComplete="off"
                  className="create-account-input"
                  id="create-input-email"
                  name="email"
                  onChange={[Function]}
                  pattern=".{1,}"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
              <p
                className="warning"
                id="email-invalid"
              >
                invalid_email_create
              </p>
            </div>
            <div
              className=""
              id="create-account-email-confirm"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-email-confirm"
                id="create-email-confirm-label"
              >
                email_confirm_field_label
                <span
                  className="asterisk"
                >
                  *
                </span>
                <input
                  autoComplete="off"
                  className="create-account-input"
                  id="create-input-email-confirm"
                  name="confirmEmail"
                  onChange={[Function]}
                  pattern=".{1,}"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
              <p
                className="warning"
              >
                invalid_email_confirmation
              </p>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-6 columns"
          >
            <div
              id="create-account-first-name"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-first-name"
                id="create-first-name-label"
              >
                first_name_field_label
                <input
                  className="create-account-input"
                  id="create-input-first-name"
                  name="firstName"
                  onChange={[Function]}
                  pattern=".{1,}"
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            className="small-6 columns"
          >
            <div
              id="create-account-last-name"
            >
              <label
                className="create-account-label"
                htmlFor="create-input-last-name"
                id="create-last-name-label"
              >
                last_name_field_label
                <input
                  className="create-account-input"
                  id="create-input-last-name"
                  name="lastName"
                  onChange={[Function]}
                  pattern=".{1,}"
                  type="text"
                  value=""
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-12 columns"
          >
            <div
              className=""
              id="create-account-password"
            >
              <div
                className="row"
              >
                <div
                  className="columns"
                >
                  <label
                    className="create-account-label"
                    htmlFor="create-input-password"
                    id="create-password-label"
                  >
                    create_password_field_label
                    <span
                      className="asterisk"
                    >
                      *
                    </span>
                    <input
                      className="create-account-input"
                      id="create-input-password"
                      name="password"
                      onChange={[Function]}
                      pattern=".{1,}"
                      required={true}
                      type="password"
                      value=""
                    />
                  </label>
                </div>
              </div>
              <p
                className="warning"
              >
                <span
                  className=""
                  id="password-length-requirement"
                >
                  password_requirements
                </span>
                <span
                  className=""
                  id="password-characters-requirement"
                >
                  password_characters_requirements
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-12 columns"
          >
            <div
              id="create-account-promotions"
            >
              <input
                checked={true}
                id="promotionsChecked"
                name="promotionsChecked"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="promotionsChecked"
              >
                hub_create_account_checkbox_promotions
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="small-12 columns"
          >
            <div
              id="create-account-privacy-container"
            >
              <p
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "account_creation_privacy_statement",
                  }
                }
                id="accept-privacy-label"
              />
            </div>
            <div
              className="row align-center"
              id="account-creation-buttons"
            >
              <div
                className="small-6 columns text-center"
              >
                <a
                  className="cancel button hollow"
                  href="/"
                  id="create-account-cancel"
                  onClick={[Function]}
                >
                  button_cancel
                </a>
              </div>
              <div
                className="small-6 columns text-center"
              >
                <button
                  className="button ghostery-button"
                  id="create-account-button"
                  type="submit"
                >
                  <span
                    className="title"
                  >
                    panel_title_create_account
                  </span>
                  <span
                    className="loader"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
